### PR TITLE
wallet: Fix for confirmed column in csv export for payment to self transactions

### LINKED ISSUE #3455: Confirmed is always true in export file
When using the export function in the transactions tab, the Confirmed status is always set to "true", even when the real status is "unconfirmed"

Example:
In transaction details you see:
Status: 0/unconfirmed
Date: 12/22/13 01:25
To: 15FJt8Upr2AADDQor1uCt8FKUEVxQTNgfp
Debit: -0.062 BTC
Transaction fee: -0.0005 BTC
Net amount: -0.0625 BTC
Transaction ID: 8ef37cd1d9a31a95fe9f0629fc83a16618b19c7ae38d7dde1ccd492b5b73a3b3

And in the csv file you see:
"Confirmed","Date","Type","Label","Address","Amount","ID"
"true","2013-12-22T01:25:53","Sent to","","15FJt8Upr2AADDQor1uCt8FKUEVxQTNgfp","-0.0625","8ef37cd1d9a31a95fe9f0629fc83a16618b19c7ae38d7dde1ccd492b5b73a3b3-000"

Good luck with this issue!

### PR DESCRIPTION
**Pull request by benthecarman**
Closes #3455

### CODE REVIEW

**Comment by laanwj** [comment]
Concept ACK.

**Comment by promag** [comment]
I'm leaning towards NAK:
 - does this fixes something? what is the point of, for instance, reading 1000 confirmations?
 - column "Type" already helps identify trusted transactions
 - this changes column order and while columns are "named" this can cause some failures if CSV is processed elsewhere

Maybe more useful is to append the block height, if any?

**Comment by jonasschnelli** [comment]
Tend to agree with @promag. Confirmation count is also probably invalid (old) the moment you export...

> [...] which can lead to transactions sent to yourself with 0 confirmations to be labeled as confirmed

Why not fix that instead?

**Comment by benthecarman** [comment]
Changed to just fix the confirmed column

File: src/qt/transactiontablemodel.cpp [hunk 1]
```diff
@@ -618,7 +618,7 @@ QVariant TransactionTableModel::data(const QModelIndex &index, int role) const
             return details;
         }
     case ConfirmedRole:
-        return rec->status.countsForBalance;
+        return (rec->status.depth > 0);
```
Comment (promag) [inline src/qt/transactiontablemodel.cpp]: Why `( )`?

**Comment by jonasschnelli** [comment]
Tested a bit.
This fixes the unconfirmed send-to-self transactions but it will list immature mined funds as confirmed.
Maybe make use of `status.status` instead of accessing the `depth`?

**Comment by jonasschnelli** [comment]
Closing due to inactivity / missing interest. Feel free to reopen.

**Comment by MarcoFalke** [comment]
@jonasschnelli This fixes a "bug". So either all (the bug and the pull) should be closed or both should be kept open (or marked for "up for grabs").

**Comment by jonasschnelli** [comment]
Oh. I missed the point that my last comments where silently addressed.

Tested ACK 9760293ce632e09f0175368ebf0c8502ac9b10d4

**Comment by MarcoFalke** [comment]
What are the steps to test for non-fluent-gui people? Does this need release notes?

**Comment by benthecarman** [comment]
> Oh. I missed the point that my last comments where silently addressed.
> 
> Tested ACK [9760293](https://github.com/bitcoin/bitcoin/commit/9760293ce632e09f0175368ebf0c8502ac9b10d4)

Sorry, should have made a comment.

**Comment by jonasschnelli** [comment]
> What are the steps to test for non-fluent-gui people?

Export a CSV?

>  Does this need release notes?

I don't think it needs an extra note (next to the PR title that usually will be listed under GUI). It's not a notable change/fix.

**Comment by MarcoFalke** [comment]
Screenshot from 2020-05-29 09-20-51

Thanks. Tested that this indeed seems to fix a bug.

**Pull request merged**